File: zd_07.py
def calculate_v2(V, S, t, V1):
    """Calculate V2 using the given formula"""
    return S / (2 * S / V - t)

def bisection_method(V, S, t, a, b, epsilon):
    """Find V2 using the bisection method within given bounds [a, b]"""
    V1 = S / t  # Calculate V1
    V2_a = calculate_v2(V, S, t, a)
    V2_b = calculate_v2(V, S, t, b)

    while abs(V2_a - V2_b) > epsilon:
        mid = (a + b) / 2
        V2_mid = calculate_v2(V, S, t, mid)

        if V2_mid * V2_a < 0:
            b = mid
            V2_b = V2_mid
        else:
            a = mid
            V2_a = V2_mid

    return (a + b) / 2

def main():
    # User input
    V = float(input("Enter average speed V: "))
    S = float(input("Enter distance S: "))
    t = float(input("Enter time t: "))

    # Calculate V1
    V1 = S / t
    print(f"Speed V1: {V1}")

    # Bisection method parameters
    a = 0  # Initial lower bound for V2
    b = 2 * V  # Initial upper bound for V2
    epsilon = 1e-6  # Precision

    # Find V2 using bisection method
    V2 = bisection_method(V, S, t, a, b, epsilon)
    print(f"Speed V2: {V2}")

if __name__ == "__main__":
    main()

File: zd_07.cpp
#include <iostream>
#include <cmath>

using namespace std;

// Function to calculate V2
double calculateV2(double V, double S, double t, double V1) {
    return S / (2 * S / V - t);
}

// Bisection method function
double bisectionMethod(double V, double S, double t, double a, double b, double epsilon) {
    double V1 = S / t; // Calculate V1
    double V2_a = calculateV2(V, S, t, a);
    double V2_b = calculateV2(V, S, t, b);

    while (fabs(V2_a - V2_b) > epsilon) {
        double mid = (a + b) / 2;
        double V2_mid = calculateV2(V, S, t, mid);

        if (V2_mid * V2_a < 0) {
            b = mid;
            V2_b = V2_mid;
        } else {
            a = mid;
            V2_a = V2_mid;
        }
    }

    return (a + b) / 2;
}

int main() {
    double V, S, t;
    cout << "Enter average speed V: ";
    cin >> V;
    cout << "Enter distance S: ";
    cin >> S;
    cout << "Enter time t: ";
    cin >> t;

    double V1 = S / t;
    cout << "Speed V1: " << V1 << endl;

    double a = 0, b = 2 * V; // Initial bounds for V2
    double epsilon = 1e-6; // Precision

    double V2 = bisectionMethod(V, S, t, a, b, epsilon);
    cout << "Speed V2: " << V2 << endl;

    return 0;
}